docs/source/en/features/lazy_init.md

```diff
@@ -0,0 +1,68 @@
+# Lazy initialization
+
+Author: Hongxin Liu
+
+**Prerequisite**
+- [Booster API](../basics/boolster_api.md)
+- [Booster Plugins](../basics/boolster_plugins.md)
+- [Booster Checkpoint](../basics/boolster_checkpoint.md)
+
+**Related discussion**
+- [Lazy initialization of model](https://github.com/hpcaitech/ColossalAI/discussions/3124)
+
+## Introduction
+
+LazyTensor allows DL framework (PyTorch) to execute operations lazily, by storing all operations related to it and reruning them when it's required to be materialized.
+
+LazyInit defers model initialization and it's based on LazyTensor.
+
+This is especially useful when we use model parallelism to train large models, in which case the model cannot fit in GPU memory. Through this, we can initialize model tensors using meta tensor and do static analysis to get shard strategy. And then materialize each tensor and apply the shard strategy. The static analysis can be omitted if the shard strategy is known in advance.
+
+## Usage
+
+You may use lazy initialization when using Gemini, tensor parallelism, pipeline parallelism, and auto-parallelism. In other cases, you may not need to use lazy initialization.
+
+Gemini is compatible with lazy initialization. You can use them together directly.
+
+<!--- doc-test-ignore-start -->
+```python
+from colossalai.booster import Booster
+from colossalai.booster.plugin import GeminiPlugin
+from colossalai.lazy import LazyInitContext
+
```

Comment:
All the code is ignored, should we test it?

Reply:
This would be tested in unit test of gemini plugin. We can also test it here. I will update.

Reply:
added